ecl/provider_connectivity/v2/tenant_connections/results.go

```diff
@@ -0,0 +1,84 @@
+package tenant_connections
+
+import (
+	"github.com/nttcom/eclcloud"
+	"github.com/nttcom/eclcloud/pagination"
+)
+
+// TenantConnection represents Tenant Connection.
+type TenantConnection struct {
+	ID                        string            `json:"id"`
+	TenantConnectionRequestID string            `json:"tenant_connection_request_id"`
+	Name                      string            `json:"name"`
+	Description               string            `json:"description"`
+	Tags                      map[string]string `json:"tags"`
+	TenantID                  string            `json:"tenant_id"`
+	NameOther                 string            `json:"name_other"`
+	DescriptionOther          string            `json:"description_other"`
+	TagsOther                 interface{}       `json:"tags_other"`
```

Comment:
Why do you user interface here instead of string map? It is not consistent with other codes.

Reply:
This is because the type of "tags_other" returned by the Create API depends on the pattern.
If the "device_type" of the request is "ECL :: Compute :: Server", it is a map type, otherwise a string is returned.

Reply:
Can you share some example of response data? And please add comments in code because it will confuse us without any note.

Reply:
Yes, this is the response data when executing tenant_connections.Create ().Look at the value of tags_other.

DeviceType : ECL::Compute::Server
```
map[tenant_connection:map[description: description_other: device_id:8416f86c-d884-4f89-8a2a-3c33b3cd9807 device_interface_id: device_type:ECL::Compute::Server id:4bc34650-494d-11ea-a27c-525400060500 name:sdktest name_other: network_id:15e79044-c6f7-4f92-981d-6ed52b9b5a54 port_id: status:creating tags:map[] tags_other:map[] tenant_connection_request_id:147c4ffa-481e-11ea-8088-525400060300 tenant_id:fa68e8d56ff7468d8fb1b056bf13bf48 tenant_id_other:d2f19c353e6d4c519e530c6a78438b33]]
```
DeviceType : ECL::Baremetal::Server
```
map[tenant_connection:map[description: description_other: device_id:0acab22f-8993-451c-8a6b-398b0244f578 device_interface_id:46eb7624-d462-46c2-8ac7-f988a15d3280 device_type:ECL::Baremetal::Server id:eb81b596-494d-11ea-a27c-525400060500 name:sdktest name_other: network_id:15e79044-c6f7-4f92-981d-6ed52b9b5a54 port_id: status:creating tags:map[] tags_other:{} tenant_connection_request_id:147c4ffa-481e-11ea-8088-525400060300 tenant_id:fa68e8d56ff7468d8fb1b056bf13bf48 tenant_id_other:d2f19c353e6d4c519e530c6a78438b33]]
```
DeviceType : ECL::VirtualNetworkAppliance::VSRX
```
map[tenant_connection:map[description: description_other: device_id:96cd2557-bd88-4fda-8737-a7c3cd3cd661 device_interface_id:interface_2 device_type:ECL::VirtualNetworkAppliance::VSRX id:728c8ac0-494e-11ea-8088-525400060300 name:sdktest name_other: network_id:15e79044-c6f7-4f92-981d-6ed52b9b5a54 port_id: status:creating tags:map[] tags_other:{} tenant_connection_request_id:147c4ffa-481e-11ea-8088-525400060300 tenant_id:fa68e8d56ff7468d8fb1b056bf13bf48 tenant_id_other:d2f19c353e6d4c519e530c6a78438b33]]
```

Comments are included to avoid confusion.

Reply:
Can you write comment in detail(e.g. data example of each type)?
Does `tags_other:{}` mean blank string? I wanna look at API response.

Reply:
Describe the API response for each DeviceType.
DeviceType : ECL::Compute::Server
```
{"tenant_connection": {"id": "65cf4752-4d57-11ea-a27c-525400060500", "tenant_id": "fa68e8d56ff7468d8fb1b056bf13bf48", "tenant_id_other": "d2f19c353e6d4c519e530c6a78438b33", "tenant_connection_request_id": "147c4ffa-481e-11ea-8088-525400060300", "network_id": "15e79044-c6f7-4f92-981d-6ed52b9b5a54", "device_type": "ECL::Compute::Server", "device_id": "8416f86c-d884-4f89-8a2a-3c33b3cd9807", "device_interface_id": "", "port_id": "", "status": "creating", "name": "", "description": "", "tags": {}, "name_other": "", "description_other": "", "tags_other": {}}}
```
DeviceType : ECL::Baremetal::Server
```
{"tenant_connection": {"id": "dcfb18be-4d5d-11ea-a27c-525400060500", "tenant_id": "fa68e8d56ff7468d8fb1b056bf13bf48", "tenant_id_other": "d2f19c353e6d4c519e530c6a78438b33", "tenant_connection_request_id": "147c4ffa-481e-11ea-8088-525400060300", "network_id": "15e79044-c6f7-4f92-981d-6ed52b9b5a54", "device_type": "ECL::Baremetal::Server", "device_id": "0acab22f-8993-451c-8a6b-398b0244f578", "device_interface_id": "46eb7624-d462-46c2-8ac7-f988a15d3280", "port_id": "", "status": "creating", "name": "", "description": "", "tags": {}, "name_other": "", "description_other": "", "tags_other": "{}"}}
```
DeviceType : ECL::VirtualNetworkAppliance::VSRX
```
{"tenant_connection": {"id": "cc72329c-4d5e-11ea-95a8-525400060400", "tenant_id": "fa68e8d56ff7468d8fb1b056bf13bf48", "tenant_id_other": "d2f19c353e6d4c519e530c6a78438b33", "tenant_connection_request_id": "147c4ffa-481e-11ea-8088-525400060300", "network_id": "15e79044-c6f7-4f92-981d-6ed52b9b5a54", "device_type": "ECL::VirtualNetworkAppliance::VSRX", "device_id": "96cd2557-bd88-4fda-8737-a7c3cd3cd661", "device_interface_id": "interface_2", "port_id": "", "status": "creating", "name": "sdktest", "description": "", "tags": {}, "name_other": "", "description_other": "", "tags_other": "{}"}}
```

> Does tags_other:{} mean blank string?

No, tags_other: {} means the string '{}'.